crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -849,3 +954,55 @@ fn is_airflow_builtin_or_provider(segments: &[&str], module: &str, symbol_suffix
         _ => false,
     }
 }
+
+fn is_task_context_referenced(checker: &mut Checker, expr: &Expr) -> bool {
+    let parents: Vec<_> = checker.semantic().current_statements().collect();
+
+    for stmt in parents {
```

Comment:
We should avoid allocating a vector here as we can directly iterate over the statement tree like so:
```rs
for stmt in checker.semantic().current_statements() {
	// ...
}
```

Reply:
removed it

Reply:
removed it